pysisyphus/cos/ChainOfStates.py

```diff
@@ -231,6 +231,38 @@ def set_images(self, indices, images):
         for ind, image in zip(indices, images):
             self.images[ind] = image
 
+    def concurrent_force_calcs(self, images_to_calculate, image_indices):
+            client = self.get_dask_client()
+            self.log(client)
+
+            # save original pals to restore them later
+            orig_pals = [self.images[i].calculator.pal for i in range(len(self.images))]
+
+            # divide pal of each image by the number of workers available or available images
+            n_workers = len(client.scheduler_info()["workers"]) # number of workers available
+            n_images = len(images_to_calculate) # number of images to calculate
+
+            #split images to calculate into batches of n_workers and set pal of each image
+            n_batches = n_images // n_workers
+            for i in range(0, n_batches):
+                for j in range(i*n_workers, (i+1)*n_workers):
+                    new_pal = images_to_calculate[j].calculator.pal // n_workers # divide pal by the number of workers
+                    images_to_calculate[j].calculator.pal = new_pal if new_pal > 0 else 1 # set pal to 1 if it is smaller than 1
```

Comment:
`new_pal` should be a constant in the scope of the function, shouldn't it? All images will/should have the same `pal` value ...
If so `new_pal` should be calculated outside of the loop
`new_pal = min(1, images_to_calculate[0].calculator.pal // n_workers)`